Celeste.Mod.mm/Patches/MapData.cs

```diff
@@ -189,20 +221,10 @@ public patch_MapData(AreaKey area)
                     CoreMode = area.CoreMode,
                     Dreaming = area.Dreaming
                 });
-            }
-
-            BinaryPacker.Element modeMeta = meta.Children?.FirstOrDefault(el => el.Name == "mode");
-            if (modeMeta == null)
-                return;
-
-            new MapMetaModeProperties(modeMeta).ApplyTo(area, mode);
-
-            // Metadata for B-Side and C-Side are parsed and stored.
-            if (mode != AreaMode.Normal) {
-                MapMeta mapMeta = new MapMeta(meta) {
-                    Modes = area.GetMeta().Modes
-                };
-                area.Mode[(int) mode].SetMapMeta(mapMeta);
+            } else {
+                MapMeta combinedMeta = MapMeta.Add(metaParsed, area.GetMeta());
+                area.Mode[(int)mode].SetMapMeta(combinedMeta);
```

Comment:
Hmmm.... I wonder if there should be a SetMapMeta call in the Normal branch as well. I can't figure out where the map mode properties meta is used.

Reply:
It looks like there's two places mode meta is stored - on the maps themselves and in the modes property of the area meta. Should we try to stick to one of them, e.g. by following the same contract that I made up for the new mode property where MapMeta.modes will be null outside of the loading context?